large-anon/src/Data/Record/Anon/Internal/Plugin/TC/Parsing.hs

```diff
@@ -152,4 +152,4 @@ parseInjTyConApp tcs tyCon t = do
     -- When this is the case, however, by injectivity of 'tyCon' we know that
     -- @args1 ~ args1'@, so we can just return /any/ of the matches; we will
     -- return the first.
-    lookup tyCon (toList splits)
+    lookup tyCon [ (x,y) | (x,y,_) <- toList splits] -- TODO: do we need coercion?
```

Comment:
As far as I understand, it is never sound to discard these coercions, because that could mean floating out evidence past an enclosing Given.